extensions/commands/sbom/cmd_cyclonedx.py

```diff
@@ -0,0 +1,183 @@
+            return [LicenseChoice(license=LicenseFactory().make_from_string(i)) for i in ls]  # noqa
+        else:
+            return [LicenseChoice(license_=LicenseFactory().make_from_string(i)) for i in ls]
+
+    def package_url(node: 'Node') -> Optional[PackageURL]:
+        """
+        Creates a PURL following https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#conan
+        """
+        return PackageURL(
+            type="conan",
+            name=node.name,
+            version=node.conanfile.version,
+            qualifiers={
+                "prev": node.prev,
+                "rref": node.ref.revision if node.ref else None,
+                "user": node.conanfile.user,
+                "channel": node.conanfile.channel,
+                "repository_url": node.remote.url if node.remote else None
+            }
+        ) if node.name else None
+
+    def create_component(node: 'Node') -> Component:
+        purl = package_url(node)
+        if cyclonedx_major_version_is_4():
+            component = Component(
+                type=package_type_to_component_type(node.conanfile.package_type),  # noqa
+                name=node.name or f'UNKNOWN.{id(node)}',
+                version=node.conanfile.version,
+                licenses=licenses(node.conanfile.license),
+                bom_ref=purl.to_string() if purl else None,
+                purl=purl,
+                description=node.conanfile.description
+            )
+        else:
+            component = Component(
+                component_type=package_type_to_component_type(node.conanfile.package_type),
+                name=node.name or f'UNKNOWN.{id(node)}',
+                version=node.conanfile.version,
+                licenses=licenses(node.conanfile.license),
+                bom_ref=purl.to_string() if purl else None,
+                purl=purl,
+                description=node.conanfile.description
+            )
+        if node.conanfile.homepage and cyclonedx_major_version_is_4():  # bug in cyclonedx 3 enforces hashes
+            component.external_references.add(ExternalReference(
+                type=ExternalReferenceType.WEBSITE,  # noqa
+                url=XsUri(node.conanfile.homepage),
+            ))  # noqa
+        return component
+
+    def me_as_tool() -> Tool:
+        tool = Tool(name="conan extension recipe:create-sbom")
+        if cyclonedx_major_version_is_4():
+            tool.external_references.add(ExternalReference(
+                type=ExternalReferenceType.WEBSITE,  # noqa
+                url=XsUri("https://github.com/conan-io/conan-extensions")))  # noqa
+        else:
+            tool.external_references.add(ExternalReference(
+                reference_type=ExternalReferenceType.WEBSITE,
+                url=XsUri("https://github.com/conan-io/conan-extensions")))
+        return tool
+
+    # region COPY FROM conan: cli/commands/graph.py
+    common_graph_args(parser)
+    args = parser.parse_args(*args)
```

Comment:
```suggestion
    # FIXME: Process the ``--build-require`` argument
    parser.add_argument("--build-require", action='store_true', default=False,
                        help='Whether the provided path is a build-require')
    args = parser.parse_args(*args)
```

This was added to the ``validate_common_graph_args()`` checks recently, but the ``common_graph_args()`` doesn't add it yet. It will probably do it, but this is part of the stabilization of the Python API for custom commands. It is expected that this will still break.